src/logging.cpp

```diff
@@ -196,29 +199,31 @@ std::string BCLog::Logger::LogTimestampStr(const std::string &str)
     } else
         strStamped = str;
 
-    if (!str.empty() && str[str.size()-1] == '\n')
-        m_started_new_line = true;
-    else
-        m_started_new_line = false;
-
     return strStamped;
 }
 
 void BCLog::Logger::LogPrintStr(const std::string &str)
 {
-    std::string strTimestamped = LogTimestampStr(str);
+    std::string strPrefixed = str;
+
+    if (m_log_threadnames && m_started_new_line)
+        strPrefixed.insert(0, "[" + std::move(thread_util::GetInternalName()) + "] ");
```

Comment:
I think this is a pessimizing move which prevents copy elision.

Reply:
Fixed, thanks.